src/input/gamepad/gamepadCalibration.js

```diff
@@ -0,0 +1,232 @@
+// @flow
+/*eslint indent:0*/
+
+import {Vec2D} from "../../main/util/Vec2D";
+
+import type {ButtonInfo, StickInfo, StickCardinals, TriggerInfo, DPadInfo, GamepadInfo} from "./gamepadInfo";
+import type {Gamepad, Button} from "./gamepad";
+
+function scanForButton ( buttons0 : Array<Button>, buttons1 : Array<Button>
+                       , axes0 : Array<number>, axes1 : Array<number>
+                       ) : null | ButtonInfo {
+  let buttonInfo = null;
```

Comment:
For instance, for buttons, we have a snapshot 0 corresponding to null input, and a snapshot 1 where the user is supposed to be inputting the desired button (and no other buttons).

I picked an approach like this rather than a continual "keep scanning for buttons until something changes", as that approach doesn't work well to detect the range of triggers and axes.

The calibration screen should have a visual indicator that a snapshot is about to be taken, and indicate when it is taken.